### PR TITLE
Ngc 2386 retrieve widget data

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

**Comment by AndyHWChung** [comment]
I've created a module which reads the HTS database entries and outputs the rows as a list of base 3 numbers. This is done once on startup and uses mongo-lock to ensure the task is carried out only in one instance of the app. I've assumed the form of the data follows the one I found in this comment I found in JIRA:

https://jira.tools.tax.service.gov.uk/browse/NGC-2264?focusedCommentId=414068&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-414068

If this isn't correct then once someone tells me what the correct format is then I can go and change it. All this code lives in the `temp` package - once the functionality is no longer needed the whole `temp` package can be deleted.

Since this whole thing is temporary I haven't bothered putting in unit tests. I have tested locally however and it works as expected.

**Comment by AndyHWChung** [comment]
just for reference, here is a function that can be used to decompose a number base `n`:

```scala
def decompose(base: Int)(x: Int): List[Int] = {
  def loop(x: Int, acc: List[Int]): List[Int] = {
    if(x == 0){
      List(0)
    } else {
      val m = x % base

      if (x < base) {
        m :: acc
      } else {
        loop((x - m)/base, m :: acc)
      }
    }
  }

  loop(x, Nil).reverse
}
```

For example:
```
(0 to 10).map{ i ⇒ i → decompose(3)(i) }.mkString("\n")
```
yields
```scala
(0,List(0))
(1,List(1))
(2,List(2))
(3,List(0, 1))
(4,List(1, 1))
(5,List(2, 1))
(6,List(0, 2))
(7,List(1, 2))
(8,List(2, 2))
(9,List(0, 0, 1))
(10,List(1, 0, 1))
```
(note that the least significant digit is on the left)

File: it/uk/gov/hmrc/nativeappwidget/controllers/SurveyWidgetDataControllerISpec.scala [hunk 1]
```diff
@@ -18,18 +18,19 @@ package uk.gov.hmrc.nativeappwidget.controllers
...
+import uk.gov.hmrc.nativeappwidget.models.{Content, KeyValuePair, SurveyData}
 import uk.gov.hmrc.nativeappwidget.repos.{SurveyWidgetMongoRepository, SurveyWidgetRepository}
 import uk.gov.hmrc.nativeappwidget.stubs.AuthStub
 import uk.gov.hmrc.nativeappwidget.support.BaseISpec
 
 import scala.collection.immutable
 import scala.concurrent.ExecutionContext.Implicits.global
 
-class SurveyWidgetDataControllerISpec extends BaseISpec with Eventually {
+class SurveyWidgetDataControllerISpec extends BaseISpec with Eventually with BeforeAndAfterEach {
```
Comment (frankoid) [inline it/uk/gov/hmrc/nativeappwidget/controllers/SurveyWidgetDataControllerISpec.scala]: I don't think the changes to this Spec are needed

**Comment by frankoid** [comment]
Looks ready to go overall, there are a few changes that don't seem necessary but look like minor improvements, I think I like the old version of SurveyWidgetDataControllerISpec better though.

File: it/uk/gov/hmrc/nativeappwidget/controllers/SurveyWidgetDataControllerISpec.scala (hunk 1 above)
Comment (AndyHWChung) [reply it/uk/gov/hmrc/nativeappwidget/controllers/SurveyWidgetDataControllerISpec.scala]: that's a remnant of the changes I made to the integration tests when I wrote the endpoint to retrieve the data and had a integration test for that. I took out that test but didn't take out the methods which I had factored out when writing those tests

**Pull request merged**